### PR TITLE
Fix/snowflake double quotes

### PR DESCRIPTION
**Pull request by Starstruckk**
### Brief summary of the change made
-- Added single quotes support for TAG object
-- Added double quotes support for columns inside row access policy

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -4570,6 +4575,17 @@ class CreateViewStatementSegment(ansi.CreateViewStatementSegment):
     )
 
 
+class ColumnQuoteSegment(BaseSegment):
+    """Double quotes column segment overriding default ColumnReferenceSegment."""
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:4579]: Why not override `ColumnReferenceSegment`? Or can this alternative only be used in certain cases?

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 2]
```diff
@@ -4827,7 +4843,7 @@ class CsvFileFormatTypeParameters(BaseSegment):
         Sequence(
             "NULL_IF",
             Ref("EqualsSegment"),
-            Bracketed(Delimited(Ref("QuotedLiteralSegment"), optional=True)),
+            OptionallyBracketed(Delimited(Ref("QuotedLiteralSegment"), optional=True)),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:4846]: Is this change supposed to be here? And, if so, can we add a test for it?

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 2 above)
Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_snowflake.py:4846]: Checking the docs, I don't think this is allowed https://docs.snowflake.com/en/sql-reference/sql/alter-file-format

I'll revert it.

**Review by alanmcruickshank — APPROVED** [review]
I've reverted a few of the controversial bits of this so that the bits we agree on can be merged.

**Pull request merged**